### PR TITLE
Added a mode to gapit to only trace a single API.

### PR DESCRIPTION
**Pull request by AWoloszyn**
This will allow us to filter out traced commands during trace.
Specifically this means we don't have to care about GLES during a
vulkan trace.

### CODE REVIEW

File: gapii/cc/call_observer.cpp [hunk 1]
```diff
@@ -112,7 +113,7 @@ void CallObserver::invoke() {
 
 void CallObserver::encodeAndDeleteCommand(::google::protobuf::Message* cmd) {
     observePending();
-    auto encoder = mSpyPtr->getEncoder();
+    auto encoder = mSpyPtr->getEncoder(mApi);
```
Comment (Qining) [inline gapii/cc/call_observer.cpp]: When we call `encodeAndDeletCommand`, what is the value of `mApi`? I think the value is `0` as in spy.cpp, the observer is constructed with `0` value. But I think the value needs to be `1` as that is the bit mask value for Core API?

File: gapii/cc/spy.cpp [hunk 2]
```diff
@@ -205,13 +205,14 @@ Spy::Spy()
...
-
+    set_valid_apis(header.mAPIs);
+    GAPID_ERROR("APIS %08x", header.mAPIs);
     GAPID_INFO("GAPII connection established. Settings:");
     GAPID_INFO("Observe framebuffer every %d frames", mObserveFrameFrequency);
     GAPID_INFO("Observe framebuffer every %d draws", mObserveDrawFrequency);
     GAPID_INFO("Disable precompiled shaders: %s", mDisablePrecompiledShaders ? "true" : "false");
 
-    CallObserver observer(this);
+    CallObserver observer(this, 0);
```
Comment (Qining) [inline gapii/cc/spy.cpp]: Is it fine that the `mApi` value of the observer is `0`? I think we need it to be `1` as we built an Architecture atom below?

File: cmd/gapit/trace.go [hunk 3]
```diff
@@ -49,12 +49,17 @@ func init() {
 	})
 }
 
+const coreAPI = uint32(1 << 0)
```
Comment (ben-clayton) [inline cmd/gapit/trace.go]: Please add a comment saying these hard-coded constants need to be kept in-sync with the `api_index` in the API files.

File: gapii/cc/call_observer.cpp (hunk 1 above)
Comment (AWoloszyn) [inline gapii/cc/call_observer.cpp]: The Call Observer is re-created for every command. This is up to date with each command that may be run. (Vulkan/Gles/Core all set it when they create the call observer).

File: gapii/cc/spy.cpp (hunk 2 above)
Comment (AWoloszyn) [inline gapii/cc/spy.cpp]: Good catch.

File: cmd/gapit/trace.go (hunk 3 above)
Comment (AWoloszyn) [inline cmd/gapit/trace.go]: Done

**Comment by Qining** [comment]
looks good for me

**Pull request merged**